rotkehlchen/db/dbhandler.py

```diff
@@ -2660,21 +2661,33 @@ def get_rotkehlchen_premium(self) -> Optional[PremiumCredentials]:
         # else
         return None
 
-    def get_netvalue_data(self, from_ts: Timestamp) -> Tuple[List[str], List[str]]:
+    def get_netvalue_data(
+        self,
+        from_ts: Timestamp,
+        include_nfts: bool = True,
+    ) -> Tuple[List[str], List[str]]:
         """Get all entries of net value data from the DB"""
         cursor = self.conn.cursor()
+        nft_balances = []
         # Get the total location ("H") entries in ascending time
         query = cursor.execute(
             f'SELECT time, usd_value FROM timed_location_data '
             f'WHERE location="H" AND time >= {from_ts} ORDER BY time ASC;',
         )
-
+        historical_data = query.fetchall()
+        if not include_nfts:
+            nft_balances = cursor.execute(
+                f'SELECT time, SUM(usd_value) FROM timed_balances WHERE time >= ? '
+                f'AND currency LIKE "{NFT_DIRECTIVE}%" GROUP BY time',
+                (from_ts,),
+            )
+        nft_values = {time: value for time, value in nft_balances}
```

Comment:
This is not wrong and will work. Just thinking a bit here. Can we optimize a bit by changing the logic up?
Idea.

Remove `nft_balances` from 2671 and make:

```suggestion
        historical_data = query.fetchall()
        if not include_nfts:
            nft_balances = cursor.execute(
                f'SELECT time, SUM(usd_value) FROM timed_balances WHERE time >= ? '
                f'AND currency LIKE "{NFT_DIRECTIVE}%" GROUP BY time',
                (from_ts,),
            )
            nft_values = {time: value for time, value in nft_balances}
```

And then in 2687:

```python
for entry in historical_data:
    times_int.append(entry[0])
    if include_nfts:
        data.append(entry[1])
    else:
        data.append(FVal(entry[1]) - FVal(nft_values.get(entry[0], 0)))
```

This way the code is faster for the most common case, which is to include the options for NFTs and probably the other too.

1. `nft_values` is only created in the case where it's needed
2. in every iteration of the historical data loop there is an extra if check, but that removes quite a bit of extra logic from  the most common case which I asume would be the current behaviour of taking nft into consideration.